fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/deployment/RetryContext.java

```diff
@@ -103,11 +103,16 @@ private boolean retryOn(Throwable throwable) {
         return Arrays.stream(retryOn).anyMatch(t -> t.isAssignableFrom(throwable.getClass()));
     }
 
-    void delayIfNeeded() throws InterruptedException {
-        if (delay > 0) {
-            long jitterBase = config.getJitter();
-            long jitter = (long) (Math.random() * ((jitterBase * 2) + 1)) - jitterBase; // random number between -jitter and +jitter
-            TimeUnit.MILLISECONDS.sleep(delay + Duration.of(jitter, config.getJitterDelayUnit()).toMillis());
+    void delayIfNeeded() {
+        try {
+            if (delay > 0) {
+                long jitterBase = config.getJitter();
+                long jitter = (long) (Math.random() * ((jitterBase * 2) + 1)) - jitterBase; // random number between -jitter and +jitter
+                TimeUnit.MILLISECONDS.sleep(delay + Duration.of(jitter, config.getJitterDelayUnit()).toMillis());
+            }
+        } catch (InterruptedException e) {
```

Comment:
Any reason why can't rethrow like we used to?

Reply:
Hm, this method is used in `nextRetry()` so we need to try-catch but I think we could return the exception instead of creating ISE. Let me try this.